### ISSUE TITLE
plex shows nothing on the page

### ISSUE THREAD

**Issue by off112a** [XDGFX/ultrasonics#24]
Plexpage shows nothing after I have setup the Plex plugin and pressed submit.

The same thing happens if I choose plex as input. 

I have reinstall the docker image but the same happens every time.

![plex](https://user-images.githubusercontent.com/22722009/117354319-7f1d4080-aeb1-11eb-85f4-d1897bb483f1.png)

**Comment by XDGFX** [comment]
Could you post any relevant logs? Hopefully it shows something isn't working when you try to submit.

**Comment by rodrigorodrigo** [comment]
can confirm that this is happening. After putting plex configs, it just goes straight to the main page. Console doesn't show anything significant.

**Comment by XDGFX** [comment]
Alright, looks like Plex is returning an xml response which isn't quite correct.

Just to confirm, you can complete and fill in this page, but clicking submit here brings you back to the homepage? Was the test successful?

![Screenshot from 2021-08-12 23-27-26](https://user-images.githubusercontent.com/16356208/129277959-50a2cf76-5c54-4fd2-a201-6604cb6e46ba.png)

I want to have a look at what XML file Plex is returning, could you please add the following line above line 366 in `ultrasonics/official_plugins/up_plex.py`?
```python
    log.error(resp.text)
```

It should look like this:

![image](https://user-images.githubusercontent.com/16356208/129278145-b02942f0-2b6e-4582-8b22-9f18b687d2a0.png)

Then run again, and it should print the whole XML file as an error to console. Copy and paste it here inside a code block and I'll have a look :)

**Comment by rodrigorodrigo** [comment]
Server URL only tests correct if I don't insert the port which, according to the placeholder syntax is not correct. 

I will log that line later today and will post the results.

**Comment by rodrigorodrigo** [comment]
some other debugging that may help:


```
Traceback (most recent call last):
File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
response = self.full_dispatch_request()
File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1953, in full_dispatch_request
return self.finalize_request(rv)
File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1968, in finalize_request
response = self.make_response(rv)
File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 2097, in make_response
raise TypeError(
TypeError: The view function did not return a valid response. The function either returned None or ended without a return statement.
[38m2021-08-16 20:32:24,247 - 🎧 plex - DEBUG - Running settings test for plugin plex v0.1 (plugins.py:187)[0m
[38m2021-08-16 20:32:24,248 - 🎧 plex - DEBUG - Checking that Plex responds to API requests... (up_plex.py:310)[0m
[38m2021-08-16 20:32:24,247 - 🎧 plex - DEBUG - Running settings test for plugin plex v0.1 (plugins.py:187)[0m
[38m2021-08-16 20:32:24,248 - 🎧 plex - DEBUG - Checking that Plex responds to API requests... (up_plex.py:310)[0m
/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py:981: InsecureRequestWarning: Unverified HTTPS request is being made to host 'plex.algumacoisa.online'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
warnings.warn(
[38m2021-08-16 20:32:24,725 - 🎧 plex - DEBUG - Test successful. (up_plex.py:317)[0m
[38m2021-08-16 20:32:24,725 - 🎧 plex - DEBUG - Testing music directories... (up_plex.py:322)[0m
[38m2021-08-16 20:32:24,725 - 🎧 plex - DEBUG - Plex and ultrasonics music paths detected successfully. (up_plex.py:324)[0m
[38m2021-08-16 20:32:24,725 - 🎧 plex - DEBUG - Checking permissions... (up_plex.py:329)[0m
[38m2021-08-16 20:32:24,726 - 🎧 plex - DEBUG - Successfully created and removed temporary ultrasonics sync directory. (up_plex.py:343)[0m
[96m2021-08-16 20:32:24,726 - 🎧 plex - INFO - Settings test successful. (up_plex.py:349)[0m
[2021-08-16 20:32:24,726] ERROR in app: Exception on /configure_plugin [POST]
```

**Comment by XDGFX** [comment]
> Server URL only tests correct if I don't insert the port which, according to the placeholder syntax is not correct.

> Unverified HTTPS request is being made to host 'plex.algumacoisa.online'

It seems that you're trying to connect to Plex over WAN. Assuming you're using a reverse proxy to forward port 443 to port 32400 on your Plex machine that's why it works without a port.

> EDIT: I'm using the docker deployment option and apparently /bin/bash is not present on the container. Any other way to access it?

There should be some shell, try accessing `/bin/sh` or `/usr/bin/sh`?
If that still doesn't work you could edit the file on your computer and then overwrite it in the container using `docker cp`.

**Comment by XDGFX** [comment]
Closing as I haven't had a response from OP. Please re-open if you're still having issues!

**Comment by koiralasandesh** [comment]
First of all, thanks for the awesome tool @XDGFX.  
I was having this issue after I updated my plex password. Every time I clicked on 'Plex', it would redirect me to the homepage. I wanted to update the plex token in ultrasonics but I looked everywhere and could not find where to. 
Finally, I whipped out sqlite editor and changed the Plex Token in the db which solved the issue for me.
I suspect you have a broken plex config somewhere in db. 
I hope you find this helpful.
